scripts/maxmind-db-downloader.js

```diff
@@ -0,0 +1,126 @@
+    var sources = require(sourceFilePath);
+    var remainingDownloads = [];
+
+    // push each file-load-function onto the remainingDownloads queue
+    for (var source in sources) {
+      var url = sources[source];
+      if (process.env.NODE_ENV !== 'test') {
+        log.info('Adding ' + url);
+      }
+      // get the file name without the extension
+      var targetFileName = path.parse(source).name;
+      var targetFilePath = path.join(targetDirPath, targetFileName);
+      remainingDownloads.push(this.download(url, targetFilePath));
+      if (process.env.NODE_ENV !== 'test') {
+        log.info('Setting ' + targetFilePath + ' as target file');
+      }
+    }
+    return remainingDownloads;
+  };
+
+  this.download = function (url, targetFilePath) {
+    // closure to separate multiple file-download
+    return function downloadFunctor(callback) {
+      var stream = request(url);
+      stream.pipe(zlib.createGunzip()).pipe(fs.createWriteStream(targetFilePath)).on('finish', function (err) {
+        // forces overwrite, even if file exists already
+        if (err) {
+          log.error(err);
+        } else {
+          // extraction is complete
+          if (process.env.NODE_ENV !== 'test') {
+            log.info('unzip complete');
+          }
+          callback();
+        }
+      });
+    };
+  };
+
+  this.startDownload = function (remainingDownloads) {
+    async.parallel(remainingDownloads, function (err) {
+      if (err) {
+        log.error(err);
+      } else {
+        if (process.env.NODE_ENV !== 'test') {
+          log.info('Downloads complete');
+        }
+      }
+    });
+    if (process.env.NODE_ENV !== 'test') {
+      log.info('Last Update: ', new Date());
+    }
+  };
+
+  this.setupAutoUpdate = function (cronTiming, remainingDownloads, timeZone) {
+    cronTiming = cronTiming || DEFAULTS.DEFAULT_CRON_TIMING;
+    timeZone = timeZone || DEFAULTS.DEFAULT_TIMEZONE;
+    var self = this;
+    // by default run periodically on Wednesday at 01:30:30.
+    new CronJob(cronTiming, function() { // eslint-disable-line no-new
+      self.startDownload(remainingDownloads);
+    }, null, true, timeZone);
+    if (process.env.NODE_ENV !== 'test') {
+      log.info('Set up auto update with cronTiming: ' + cronTiming);
+    }
+  };
+};
+
+if (require.main === module) {
+  // start download only when script is
+  // executed, not loaded through require.
+  var maxmindDbDownloader = new MaxmindDbDownloader();
+  var targetDirPath = maxmindDbDownloader.createTargetDir('db');
+  var remainingDownloads = maxmindDbDownloader.setupDownloadList(
+    path.join(__dirname, '..','sources.json'),
+    targetDirPath
+  );
+  maxmindDbDownloader.startDownload(remainingDownloads);
+  // By default, we do not setup autoUpdate, needs to be
+  // done through options when library is imported, or manually
+  // by running npm run-scripts update
+  // maxmindDbDownloader.setupAutoUpdate('30 30 1 * * 3', remainingDownloads);
```

Comment:
Not sure how we want the `setupAutoUpdate` called, need clarification on whether the user is allowed to configure this when importing the `fxa-geodb` using require (this might not be a good idea, as the cron job will stop executing when the script that called it stops execution), or whether user will run a special script e.g. `npm run-scripts auto-update` to set up the cron job, where auto-update will have the above code.
If user is going to run this from command-line, then what options are allowed to be customized/configured?

@shane-tomlinson 
@vladikoff